iXBRLViewerPlugin/__init__.py

```diff
@@ -81,19 +81,43 @@ def generateViewer(cntlr, saveViewerFile, viewerURL=DEFAULT_VIEWER_PATH, showVal
     if modelXbrl.modelDocument.type not in (Type.INLINEXBRL, Type.INLINEXBRLDOCUMENTSET):
         cntlr.addToLog("No inline XBRL document loaded.")
         return
+    copyScriptPath = None
+    if isinstance(saveViewerFile, str):
+        # Note on URLs: Rather than rely on logic to determine if the input is a file
+        # path or web address, we can allow web addresses to be considered relative paths.
+        # Unless the URL happens to resolve to an existing file on the local filesystem,
+        # it will skip this step and pass through into the viewer as expected.
+        if os.path.isabs(viewerURL):
+            viewerAbsolutePath = viewerURL
+        else:
+            viewerAbsolutePath = getAbsoluteViewerPath(saveViewerFile, viewerURL)
+
+        if os.path.isfile(viewerAbsolutePath):
```

Comment:
Does this also cover the `iXBRLViewerMenuCommand` code path?

Reply:
I hadn't accounted for that. I'll reroute that path to go through `generateViewer`. I'll also look for any other paths that may need similar treatment.

Reply:
Only other candidate I found was `/samples/build-viewer.py` which I wasn't familiar with but appears to be an alternative independent entry point for generating viewers.